File: python/leetcode/905. Sort Array By Parity.py
class Solution:
    def sortArrayByParity(self, A: List[int]) -> List[int]:
        pos = 0

        for i in range(len(A)):
            if A[i] % 2 == 0:
                temp = A[pos]
                A[pos] = A[i]
                A[i] = temp
                pos += 1
        return A

File: c++/leetcode/905. Sort Array By Parity.cpp
class Solution
{
public:
  vector<int> sortArrayByParity(vector<int> &A)
  {
    int pos = 0;

    for (int i = 0; i < A.size(); i++)
    {
      if (A[i] % 2 == 0)
      {
        int temp = A[pos];
        A[pos] = A[i];
        A[i] = temp;
        pos++;
      }
    }

    return A;
  }
};